frontend/src/App.js

```diff
@@ -1,48 +1,73 @@
-import "./App.css";
-import Dropdown from "./components/Dropdown";
-import fetchAllRoutes from "./utils/fetchAllRoutes";
-import { useEffect, useState } from "react";
+import './App.css';
+import Dropdown from './components/Dropdown';
+import fetchAllRoutes from './utils/fetchAllRoutes';
+import { useEffect, useState } from 'react';
+import ClusterCard from './components/ClusterCard';
 
 function App() {
   const [allRoutes, setAllRoutes] = useState([]);
   const [selectedRoute, setSelectedRoute] = useState('');
   const [routeDetails, setRouteDetails] = useState(null);
 
   useEffect(() => {
-    const fetch = async () => {
+    const fetchRoutes = async () => {
       try {
         const fetchedRoutes = await fetchAllRoutes();
         setAllRoutes(fetchedRoutes);
       } catch (error) {
-        console.error("Failed to fetch routes:", error);
+        console.error('Failed to fetch routes:', error);
       }
     };
-    fetch();
+
+    fetchRoutes();
   }, []);
 
   useEffect(() => {
-    if (selectedRoute) {
-      const route = allRoutes.find(route => route.routeId === parseInt(selectedRoute, 10));
-      setRouteDetails(route);
-    }
+    const fetchRouteDetails = async () => {
+      if (selectedRoute) {
+        try {
+          const route = allRoutes.find((route) => route.routeId === parseInt(selectedRoute, 10));
+          setRouteDetails(route);
+        } catch (error) {
+          console.error('Failed to fetch route details:', error);
+        }
+      }
+    };
+
+    fetchRouteDetails();
   }, [selectedRoute, allRoutes]);
 
   return (
     <div className="App">
       <h1>Route Details</h1>
-      <Dropdown 
+      <Dropdown
         routes={allRoutes}
         selectedRoute={selectedRoute}
         setSelectedRoute={setSelectedRoute}
         showId={false}
       />
-      {routeDetails && (
+      {routeDetails ? (
         <div>
-          <h2>{routeDetails.name}</h2>
+          <p>Name: {routeDetails.name}</p>
           <p>ID: {routeDetails.routeId}</p>
-          <p>Sequence: {routeDetails.sequence}</p>
-          <p>Object: {routeDetails.object}</p>
+          <p>Technical Level: {routeDetails.technicalLevelName || 'N/A'}</p>
+          <div className="clusters">
+            {routeDetails.clusters && routeDetails.clusters.length > 0 ? (
+              routeDetails.clusters.map((cluster) => (
+                <ClusterCard
+                  key={cluster.id}
+                  name={cluster.name}
+                  description={cluster.description}
+                  technicalLevelName={cluster.technicalLevelName || 'N/A'}
+                />
+              ))
+            ) : (
+              <p></p>
+            )}
+          </div>
```

Comment:
why would we want to show an empty `<p>` in the page?

If there are no clusters we just don't show anything. Remove this evaluation.

Right now you are using a [Ternary Operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator) but what you need is a [Logical AND Operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND)

Reply:
This is not resolved yet. Please replace that ternary for a Logical AND